ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlySqlLedger.scala

```diff
@@ -65,9 +67,25 @@ private[platform] object ReadOnlySqlLedger {
 
     override def acquire()(implicit context: ResourceContext): Resource[ReadOnlySqlLedger] = {
       val ledgerEndCache = MutableLedgerEndCache()
+      val storageBackendFactory = StorageBackendFactory.of(DbType.jdbcType(jdbcUrl))
       for {
-        ledgerDao <- ledgerDaoOwner(servicesExecutionContext, errorFactories, ledgerEndCache)
+        dbDispatcher <- DbDispatcher
+          .owner(
+            dataSource =
+              storageBackendFactory.createDataSourceStorageBackend.createDataSource(jdbcUrl),
+            serverRole = serverRole,
+            connectionPoolSize = databaseConnectionPoolSize,
+            connectionTimeout = databaseConnectionTimeout,
+            metrics = metrics,
+          )
           .acquire()
+        ledgerDao = ledgerDaoOwner(
```

Comment:
The method should be renamed as it no longer returns a `ResourceOwner`:
```suggestion
        ledgerDao = ledgerDao(
```

Reply:
good point, will do in upcoming commits (to gain some speed now)